### PR TITLE
Add instructions for building docker image locally to README.md

### LINKED ISSUE #8442: `make build docker` failed with "COPY failed: file not found in build context or excluded by .dockerignore: stat .build/linux-amd64/prometheus: file does not exist"
I'm on a Mac, I guess I'm holding it wrong, but I can't figure this one out...

```
make build docker
>> writing assets
# Un-setting GOOS and GOARCH here because the generated Go code is always the same,
# but the cached object code is incompatible between architectures and OSes (which
# breaks cross-building for different combinations on CI in the same container).
cd web/ui && GO111MODULE=on GOOS= GOARCH= go generate -x -v  -mod=vendor
doc.go
go run -mod=vendor assets_generate.go
ui.go
>> building binaries
GO111MODULE=on /Users/twilkie/go/bin/promu build --prefix /Users/twilkie/Documents/repos/prometheus 
 >   prometheus
 >   promtool
>> bundling npm licenses
rm -f "npm_licenses.tar.bz2"
find web/ui/react-app/node_modules -iname "license*" | tar cfj "npm_licenses.tar.bz2" --transform 's/^/npm_licenses\//' --files-from=-
docker build -t "prom/prometheus-linux-amd64:master" \
		-f ./Dockerfile \
		--build-arg ARCH="amd64" \
		--build-arg OS="linux" \
		./
Sending build context to Docker daemon  742.3MB
Step 1/22 : ARG ARCH="amd64"
Step 2/22 : ARG OS="linux"
Step 3/22 : FROM quay.io/prometheus/busybox-${OS}-${ARCH}:latest
 ---> 14d98f19c979
Step 4/22 : LABEL maintainer="The Prometheus Authors <prometheus-developers@googlegroups.com>"
 ---> Using cache
 ---> d323d10c7b66
Step 5/22 : ARG ARCH="amd64"
 ---> Using cache
 ---> 70d7ff8e87dd
Step 6/22 : ARG OS="linux"
 ---> Using cache
 ---> 96a2eede2dd4
Step 7/22 : COPY .build/${OS}-${ARCH}/prometheus        /bin/prometheus
COPY failed: file not found in build context or excluded by .dockerignore: stat .build/linux-amd64/prometheus: file does not exist
make: *** [common-docker-amd64] Error 1
```

### PR DESCRIPTION
**Pull request by tomwilkie**
Fixes #8442

Signed-off-by: Tom Wilkie <tom.wilkie@gmail.com>

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -91,9 +91,18 @@ The Makefile provides several targets:
   * *test-short*: run the short tests
   * *format*: format the source code
   * *vet*: check the source code for common errors
-  * *docker*: build a docker container for the current `HEAD`
   * *assets*: build the new experimental React UI
 
+### Building the Docker image
+
+The `make docker` target is designed for use in our CI systemd.
```
Comment (roidelapluie) [inline README.md:98]:
```suggestion
The `make docker` target is designed for use in our CI system.
```

File: README.md [hunk 2]
```diff
@@ -91,9 +91,18 @@ The Makefile provides several targets:
...
+You can build a docker image locally with the following commands:
+
+    $ promu crossbuild -p linux/amd64
```
Comment (roidelapluie) [inline README.md:101]: you still need to build the react assets

Comment (tomwilkie) [reply README.md:101]: When I ran `promu crossbuild -p linux/amd64` it did it for me:

```
➜  prometheus git:(master) promu crossbuild -p linux/amd64
> running the base builder docker image
Unable to find image 'quay.io/prometheus/golang-builder:1.15-base' locally
1.15-base: Pulling from prometheus/golang-builder
62dfdff96a79: Pull complete 
79deb48d05f4: Pull complete 
afe63264fea1: Pull complete 
0187e2a6ae46: Pull complete 
067d7b40510c: Pull complete 
46aeee8f0bd1: Pull complete 
Digest: sha256:e1306241a24129abf01ee2c81ed1003db8b3cd91c738228ac1661cf162ad1a09
Status: Downloaded newer image for quay.io/prometheus/golang-builder:1.15-base
f34d5332fa7b7214b713d0768d183b1e8f1bae6691c615601a605ea56d833c77
# linux-amd64
>> building React app
building React app
yarn run v1.22.5
$ react-scripts build
Creating an optimized production build...
Compiled successfully.

File sizes after gzip:

  285.88 KB  build/static/js/2.14ef932b.chunk.js
  30.64 KB   build/static/js/main.032f0151.chunk.js
  23.98 KB   build/static/css/2.300141ce.chunk.css
  1.5 KB     build/static/css/main.0b010d50.chunk.css
  769 B      build/static/js/runtime-main.7e00e584.js

The project was built assuming it is hosted at ./.
You can control this with the homepage field in your package.json.

The build folder is ready to be deployed.

Find out more about deployment here:

  bit.ly/CRA-deploy

Done in 39.74s.
```

Comment (roidelapluie) [reply README.md:101]: you probably want to tell how to get promu (make promu?)

**Comment by roidelapluie** [comment]
Should we have a dedicated make target instead ? That would be easier and deal with getting promu as well.

**Comment by tomwilkie** [comment]
> Should we have a dedicated make target instead ? That would be easier and deal with getting promu and the assets as well.

I'm going to say for now lets start with docs, and make a target for this later...

**Comment by roidelapluie** [comment]
Thanks for this, this is ready to be merged once DCO is fixed, maybe squash into one commit.

**Comment by roidelapluie** [comment]
Thanks!!

**Pull request merged**